ts/src/okx.ts

```diff
@@ -6736,6 +6737,38 @@ export default class okx extends Exchange {
         return result;
     }
 
+    async fetchUnderlyingAssets (params = {}) {
+        /**
+         * @method
+         * @name okx#fetchUnderlyingAssets
+         * @description fetches the market ids of underlying assets for different derivative types
+         * @see https://www.okx.com/docs-v5/en/#public-data-rest-api-get-underlying
+         * @param {object} [params] exchange specific params
+         * @returns {object[]} a list of [underlying assets]{@link https://github.com/ccxt/ccxt/wiki/Manual#underlying-assets-structure}
+         */
+        await this.loadMarkets ();
+        let type = undefined;
+        [ type, params ] = this.handleMarketTypeAndParams ('fetchUnderlyingAssets', undefined, params);
```

Comment:
@Dan-krm actually not sure anymore, at least we should default to a valid type otherwise if the user calls it without any argument it will crash

```
✗ p okx fetchUnderlyingAssets --sandbox
Python v3.10.9
CCXT v4.0.84
okx.fetchUnderlyingAssets()
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 239, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 216, in main
    result = await method(*args)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/okx.py", line 6304, in fetch_underlying_assets
    response = await self.publicGetPublicUnderlying(self.extend(request, params))
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 1839, in request
    return await self.fetch2(path, api, method, params, headers, body, config)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 1836, in fetch2
    return await self.fetch(request['url'], request['method'], request['headers'], request['body'])
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 221, in fetch
    self.handle_errors(http_status_code, http_status_text, url, method, headers, http_response, json_response, request_headers, request_body)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/okx.py", line 6353, in handle_errors
    self.throw_exactly_matched_exception(self.exceptions['exact'], code, feedback)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 2202, in throw_exactly_matched_exception
    raise exact[string](message)
ccxt.base.errors.BadRequest: okx {"code":"51000","data":[],"msg":"Parameter instType error"}
okx requires to release all resources with an explicit call to the .close() coroutine. If you are using the exchange instance with async coroutines, add `await exchange.close()` to your code into a place when you're done with the exchange and don't need the exchange instance anymore (at the end of your async coroutine).
Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x109254a00>
Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x109242980>, 87206.711795375), (<aiohttp.client_proto.ResponseHandler object at 0x109242da0>, 87206.794398833), (<aiohttp.client_proto.ResponseHandler object at 0x1092433a0>, 87206.880168208), (<aiohttp.client_proto.ResponseHandler object at 0x1092410c0>, 87207.364742125)]']
```

I think we can remove the type (if the exchange supports only one) and add that information to the docs. 

What do you think?

Reply:
I added a `marketType` argument to the method signature and updated the manual. I defaulted the marketType to 'option' and added some NotSupported error checks. Let me know if you're okay with this approach, and let me know if you think we should default the marketType in the signature, or if you have any other thoughts